### ISSUE TITLE
Window List - stretched buttons on vertical panel

### ISSUE THREAD

**Issue by usv299792458** [mate-desktop/mate-panel#269]
In Gnome 2 I used to have this layout.
![gnome](https://cloud.githubusercontent.com/assets/1787194/5520527/93f9f95a-89ae-11e4-830b-1f6afe95a8ee.png)

Unfortunately, in MATE panel, in Window List, window buttons are stretched vertically.
![mate](https://cloud.githubusercontent.com/assets/1787194/5520528/9a8f63cc-89ae-11e4-8ca0-e5b896ac7a23.png)

My setup:
MATE version 1.8.1 (Debian Jessie).
Steps to reproduce:
Select TraditionalOK theme.
Remove all panels except one, make it right-sided, set width to 30 pixels, run an application.
##

**Comment by mihai-bica** [comment]
I had the same problem and I run "mate-panel --reset" (which resets the panel layout) and the vertical windows list mate applet got fixed. I think that this is a configuration problem.

**Comment by usv299792458** [comment]
Hello mihai-bica,
Thanks for sharing your results.
Unfortunately,
> mate-panel --reset
does not fix the issue for me: it just makes top and bottom panels as by default. When I reconfigure right panel, buttons are stretched again.
I also noticed that windows are reorganized from one-column to two-column layout (in fast intervals) when is panel is on the right, and this causes "trembling" of window buttons, which makes them completely unusable.
Igor.

**Comment by mihai-bica** [comment]
Hi, you are right. mate-panel --reset does not fix the problem.

However, I downloaded the source code and did this modification:
mihai@mihai-vb ~/git/mate-panel $ git diff
diff --git a/applets/wncklet/window-list.c b/applets/wncklet/window-list.c
index 721aef2..53db860 100644
--- a/applets/wncklet/window-list.c
+++ b/applets/wncklet/window-list.c
@@ -126,6 +126,7 @@ static void applet_change_orient(MatePanelApplet* applet, MatePanelAppletOrient
        wnck_tasklist_set_orientation (tasklist->tasklist, new_orient);
 #endif
 #endif
+       wnck_tasklist_set_orientation (tasklist->tasklist, new_orient); //MIHAI
        tasklist_update(tasklist);
 }
 
@@ -451,6 +452,8 @@ gboolean window_list_applet_fill(MatePanelApplet* applet)
 #endif
 #endif
 
+       wnck_tasklist_set_orientation (tasklist->tasklist, tasklist->orientation); //MIHAI
+
        wnck_tasklist_set_icon_loader(WNCK_TASKLIST(tasklist->tasklist), icon_loader_func, tasklist, NULL);
 
        g_signal_connect(G_OBJECT(tasklist->tasklist), "destroy", G_CALLBACK(destroy_tasklist), tasklist);
mihai@mihai-vb ~/git/mate-panel $ 


and built it and the pannel got fixed. I am having a fixed panel right now on my linux mint 17 mate.

Sorry for just pasting the code here. I don't have time to commit it ... maybe I will commit at the end of this week.

Thanks.


![screenshot0001](https://cloud.githubusercontent.com/assets/7090338/7342033/cb9739d0-ecb5-11e4-95f1-96eefa1e822e.png)

**Comment by usv299792458** [comment]
Hi, again.
Hope your fix will get to debian testing soon.
There is one more problem with window list. I'm not sure if it's better to open a separate issue about it. Anyway, did you happen to observe "trembling" of window buttons sometimes, when the panel is on the side? (don't know when and for what reason it happens) It seems that buttons are reorganized to one-column and two-column layout in fast intervals and it causes this ugly effect, which makes window list completely unusable.

**Comment by mihai-bica** [comment]
Hi, I observed the trembling when I had many windows open and it completely
disappeared after this fix. Now I can have like 100 windows shown in
taskbar without problem both in horizontal and vertical orientation.

However, I don't know how it behaves on other platforms. I tested only on
mint mate 17.

On Apr 27, 2015 12:16, "usv299792458" <notifications@github.com> wrote:
>
> Hi, again.
> Hope your fix will get to debian testing soon.
> There is one more problem with window list. I'm not sure if it's better
to open a separate issue about it. Anyway, did you happen to observe
"trembling" of window buttons sometimes, when the panel is on the side?
(don't know when and for what reason it happens) It seems that buttons are
reorganized to one-column and two-column layout in fast intervals and it
causes this ugly effect, which makes window list completely unusable.

**Comment by usv299792458** [comment]
I will test on Debian 8 and write you my results in a few days.

**Comment by usv299792458** [comment]
Hello Mihai,
Thank you.
Confirm that your patch works perfectly in Debian 8.
Igor.

**Comment by mihai-bica** [comment]
Hi, I'm glad to hear this. What would be the next step? Do I have to commit this code fix?

I refactored a bit the code. Removed the `ifdef WNCK_CHECK_VERSION`. I searched for WNCK_CHECK_VERSION and it wasn't defined anywhere.

mihai@mihai-vb ~/git/mate-panel $ git diff
```
diff --git a/applets/wncklet/window-list.c b/applets/wncklet/window-list.c
index 721aef2..78a47d2 100644
--- a/applets/wncklet/window-list.c
+++ b/applets/wncklet/window-list.c
@@ -121,11 +121,7 @@ static void applet_change_orient(MatePanelApplet* applet, MatePanelAppletOrient
 
        tasklist->orientation = new_orient;
 
-#ifdef WNCK_CHECK_VERSION
-#if WNCK_CHECK_VERSION (3, 4, 6)
        wnck_tasklist_set_orientation (tasklist->tasklist, new_orient);
-#endif
-#endif
        tasklist_update(tasklist);
 }
 
@@ -445,11 +441,7 @@ gboolean window_list_applet_fill(MatePanelApplet* applet)
        tasklist->tasklist = wnck_tasklist_new(NULL);
 #endif
 
-#ifdef WNCK_CHECK_VERSION
-#if WNCK_CHECK_VERSION (3, 4, 6)
        wnck_tasklist_set_orientation (tasklist->tasklist, tasklist->orientation);
-#endif
-#endif
 
        wnck_tasklist_set_icon_loader(WNCK_TASKLIST(tasklist->tasklist), icon_loader_func, tasklist, NULL);
 ```
mihai@mihai-vb ~/git/mate-panel $  

Thanks,
Mihai

**Comment by usv299792458** [comment]
I would advise you to contact the developers.

**Comment by usv299792458** [comment]
Hello mihai-bica,

Did you write to Debian mate-panel maintainers that they remove '#ifdef WNCK_CHECK_VERSION' around 'wnck_tasklist_set_orientation' ?

BTW, did you see this? I have stumbled upon it just now:

https://github.com/mate-desktop/mate-panel/issues/58

**Comment by mihai-bica** [comment]
Hi usv299792458, I didn't write to Debian pate-panel maintainers. If you have the time, can you please ask them to remove the '#ifdef WNCK_CHECK_VERSION  (3, 4, 6)' around 'wnck_tasklist_set_orientation' ? (I don't know who to email/ask and don't have time to search.)

'BTW, did you see this? I have stumbled upon it just now: #58', no I didn't see it. I also know that it was working in the past back to gnome 2 and it got broken recently around mint 17. The bug was introduced by adding this 'ifdef' and not upgrading 'libwnck' to version 3.4.6 in Debian.

Upgrading 'libwnck' to version 3 could be problematic because it is based on GTK 3 which is experimental and that will not happen soon (see latest comment https://github.com/mate-desktop/mate-panel/pull/300).

IMHO, the best fix for us, (until we will have GTK3 mate support) is to revert the 'ifdef' in the Debian mate-panel.

**Comment by usv299792458** [comment]
This issue is posted to Debian BTS: https://bugs.debian.org/785272

**Comment by usv299792458** [comment]
Hi mihai-bica,

In case you have some information for Debian mate-panel maintainers, simply send it to 785272@bugs.debian.org

**Comment by mckaygerhard** [comment]
i discovered that when i resising the panel to inpar size numbers, the icons list dont filker, but in numeric par size filck always for same amount of  windows list, lest ilustrate:

i have 8 windows opened in the mate desktop

the panel are in 42 pixels, so windows will got crazy if i open one more program, but if panel has 82 as size, the new window willplace in new line up to down .. event are more space available:
![panel-two-windowslist-line](https://cloud.githubusercontent.com/assets/1571724/23105399/ea68f0c8-f6b4-11e6-9530-88f0a639bd55.png)

**Comment by monsta** [comment]
This should be fixed now in 1.18 because of migration to GTK+3.
